sierra2mlir/src/libfuncs/mod.rs

```diff
@@ -2420,4 +2433,283 @@ impl<'ctx> Compiler<'ctx> {
+                    PositionalArg { loc: 1, ty: SierraType::Simple(felt_type) },
+                ],
+                return_types: vec![
+                    PositionalArg { loc: 0, ty: dict_type.clone() },
+                    PositionalArg { loc: 1, ty: entry_type.clone() },
+                ],
+            },
+        );
+
+        self.create_function(
+            &id,
+            vec![block],
+            &[dict_type.get_type(), entry_type.get_type()],
+            FnAttributes::libfunc(false, true),
+        )
+    }
+
+    /// felt252_dict_entry_finalize<T> (dict_with_entry, value) -> dict
+    pub fn create_libfunc_felt252_dict_entry_finalize(
+        &'ctx self,
+        func_decl: &LibfuncDeclaration,
+        storage: &mut Storage<'ctx>,
+    ) -> Result<()> {
+        let id = func_decl.id.debug_name.as_ref().unwrap().to_string();
+        let entry_type = storage
+            .types
+            .get(&get_type_id(&func_decl.long_id.generic_args[0]))
+            .cloned()
+            .expect("type to exist");
+
+        let dict_type = SierraType::create_dict_type(self, entry_type.clone());
+        // (key (always felt), value (T), is_used (bool))
+        let entry_struct_type = self
+            .llvm_struct_type(&[self.felt_type(), entry_type.get_type(), self.bool_type()], false);
+
+        let block = self.new_block(&[dict_type.get_type(), entry_type.get_type()]);
+        let dict_value: Value = block.argument(0)?.into();
+        let value: Value = block.argument(1)?.into();
+
+        let op = self.op_const(&block, "1", self.bool_type());
+        let true_const = op.result(0)?.into();
+
+        let op = self.call_dict_get_entry_ptr(&block, dict_value, &dict_type, storage)?;
+        let entry_ptr = op.result(0)?.into();
+
+        let op = self.op_llvm_gep(&block, &[0, 1], entry_ptr, entry_struct_type)?;
+        let entry_value_ptr: Value = op.result(0)?.into();
+        let op = self.op_llvm_gep(&block, &[0, 2], entry_ptr, entry_struct_type)?;
+        let entry_is_used_ptr: Value = op.result(0)?.into();
+
+        self.op_llvm_store(&block, true_const, entry_is_used_ptr)?;
+        self.op_llvm_store(&block, value, entry_value_ptr)?;
+
+        let null_ptr_const_op = self.op_llvm_nullptr(&block);
+        let dict_value_op = self.call_dict_set_entry_ptr(
+            &block,
+            dict_value,
+            null_ptr_const_op.result(0)?.into(),
+            &dict_type,
+            storage,
+        )?;
+        let dict_value = dict_value_op.result(0)?.into();
+
+        self.op_return(&block, &[dict_value]);
+
+        storage.libfuncs.insert(
+            id.clone(),
+            SierraLibFunc::Function {
+                args: vec![
+                    PositionalArg { loc: 0, ty: dict_type.clone() },
+                    PositionalArg { loc: 1, ty: entry_type },
+                ],
+                return_types: vec![PositionalArg { loc: 0, ty: dict_type.clone() }],
+            },
```

Comment:
Since there's no implicit maybe you could use `SierraLibFunc::create_function_all_args`?